tests/ui/doc/needless_doctest_main.rs

```diff
@@ -19,4 +17,98 @@
 /// ```
 fn foo() {}
 
+#[rustfmt::skip]
+/// Description
+/// ```rust
+/// fn main() {
+//~^ error: needless `fn main` in doctest
+///     let a = 0;
+/// }
+/// ```
+fn mulpipulpi() {}
+
+#[rustfmt::skip]
+/// With a `#[no_main]`
+/// ```rust
+/// #[no_main]
+/// fn a() {
+///     let _ = 0;
+/// }
+/// ```
+fn pulpimulpi() {}
+
+// Without a `#[no_main]` attribute
+/// ```rust
+/// fn a() {
+///     let _ = 0;
+/// }
+/// ```
+fn plumilupi() {}
+
+#[rustfmt::skip]
+/// Additional function, shouldn't trigger
+/// ```rust
+/// fn additional_function() {
+///     let _ = 0;
+///     // Thus `fn main` is actually relevant!
+/// }
+/// fn main() {
+///     let _ = 0;
+/// }
+/// ```
```

Comment:
(Non blocking question) Out of curiosity, do you know why we don't want to lint this case? Nested functions are allowed and so the function would instead just be part of the implicit main fn, so it could still be removed as far as I can tell?

Reply:
Not really, I suppose it's just a sensible thing to do. When having more than one function omitting `fn main` makes it look like some pseudo-language for scripting or something.

```rust
fn a() -> bool {
    return true
}

let _ = 0;
let _ = a();
// What's the scope for this??
```